feat: feature/026

gonmarfer2 opened the pull request:
Added Census Groups to filter Census easily.
Closes #26

gonmarfer2 committed:
feat: added CensusGroups

Remade the model to add groups of censuses to filter with.

gonmarfer2 committed:
feat: intermediate commit for census views

Tried to make some views, permissions not working

gonmarfer2 committed:
feat: census groups

Modified views, model and other relevant files.

gonmarfer2 committed:
test,fix: census groups

Also fixed the views for Census and CensusGroup to add missing fragments.

AntoCampos12 commented:
Existe un problema con el POST de un censo, adjunto foto
![ERROR](https://user-images.githubusercontent.com/80269634/201137491-7e31d761-9627-432e-a1d6-e9b80ffb5584.png)

gonmarfer2 committed:
fix: groups api works fine now

I had to redo everything because I didn't properly understand how everything works

gonmarfer2 committed:
fix: changed permission class to IsAdminUser

Because the one given doesn't work (request.auth is always None). This one seems equivalent.

gonmarfer2 committed:
fix: census can now be pushed without a group

Bug found on the post method.

gonmarfer2 commented:
> Existe un problema con el POST de un censo, adjunto foto. Probar con voting_id 1, voter_id 1 ambos existiendo en bd ![ERROR](https://user-images.githubusercontent.com/80269634/201137491-7e31d761-9627-432e-a1d6-e9b80ffb5584.png)

El problema viene del POST que se ha hecho, los parámetros por defecto no sirven, hay que hacer un post con un JSON personalizado del tipo:
```json
{
    "voting_id": 2,
    "voters": [1,2,3],
    "group": {
        "name": "Grupo 1"
    }
}
```
Debe existir ya un grupo para que funcione, o no poner ninguno.

gonmarfer2 committed:
test: redone some tests to make them work again

Changed JSON data structure to avoid failures and changed parameters.

AntoCampos12 reviewed (approved):
Esta nueva versión funciona correctamente : )
